test/unit/primitives_spec.js

```diff
@@ -380,6 +380,44 @@ describe("primitives", function () {
         "Global font three",
       ]);
     });
+
+    it("should set the values if they're as expected", function () {
+      const dict = new Dict();
+      dict.set("key", "value");
+
+      dict.setIfNotExists("key", "new value");
```

Comment:
Let's also add a test for the happy flow here, so calling `setIfNotExists` with a non-existent key.